src/pymodaq/daq_utils/daq_utils.py

```diff
@@ -442,11 +442,37 @@ def uncapitalize(string, Nfirst=1):
     return string[:Nfirst].lower() + string[Nfirst:]
 
 
-def get_data_dimension(arr, scan_type='scan1D', remove_scan_dimension=False):
-    dimension = len(arr.shape)
-    if dimension == 1:
-        if arr.size == 1:
-            dimension = 0
+def get_data_dimension(data: np.ndarray, scan_type='scan1D', remove_scan_dimension=False):
```

Comment:
You can add typehints for each parameter and a returntype:

```python
from typing import Tuple

def get_data_dimension(...) -> Tuple[Tuple, str, int]:
    ...
```

Reply:
ok do it

Reply:
Done in 7295cf508c38e393feeafbf1fa16008e7fa711f6.